docs/ethereum_attestation.md

```diff
@@ -0,0 +1,133 @@
+    }
+    ...
+}
+```
+
+## Ethereum smart contract
+
+A new Ethereum smart contract is deployed to the network,
+which lets users claim their Link identities:
+
+```solidity
+contract Claims {
+    event Claimed(address indexed addr);
+    function claim(uint256 format, bytes calldata payload);
+}
+```
+
+To claim an identity, call `claim` using your Ethereum account and pass your Link identity root.
+It will emit an event `Claimed`, which later can be queried to discover your attestation.
+The claims have no expiration date and don't need to be renewed.
+
+Every new claim invalidates previous ones made with the same account.
+To revoke a claim without creating a new one, claim root hash `0`,
+which is guaranteed to not match any existing identity.
+
+Currently supported `format` values:
+- `1` - `payload` is exactly 20 bytes and contains an SHA-1 Radicle Identity root hash
+- `2` - `payload` is exactly 32 bytes and contains an SHA-256 Radicle Identity root hash
+
+We need to deploy an official instance of the `Claims` smart contract and
+it must be used by all the users.
+If anybody makes a claim using a different instance, it won't be recognized by others.
+
+## Creation of an attestation
+
+You need to perform 2 actions in any order:
+- Add or update an `https://radicle.xyz/ethereum/claim/v1` entry in your identity doc.
+The entry's `address` must be your Ethereum address.
+It's highly recommended to set a short expiration date as Ethereum claims don't expire.
+- Call `claim` in the `Claims` smart contract. The `root` must point to your link identity.
+
+## Discovery from an Ethereum address
+
+When you know an Ethereum address, you can find the claimed link identity using an Ethereum client.
+The example calls are based on the standard [client JSON RPC API][rpc] and should be exposed
+by your favourite Ethereum client library.
+It's important that the client must be trusted not to hide the events.
+
+- Use [getLogs][rpc-logs] to get the newest `Claimed` event filtered for the given ethereum address
+- Get the event's `transactionHash` field and use it to fetch the transaction which emitted it with
+[getTransactionByHash][rpc-tx]
+- Validate that the transaction signature matches its data and the ethereum address.
+For reference the signature payload content is listed [here][rpc-sign].
+- Read the Link identity root hash from the transaction data
+- Verify that the Link identity doc claims back the Ethereum address,
+see [Discovery from a Link identity](#discovery-from-a-link-identity)
+
+## Discovery from a Link identity
+
+When you know a Link identity, you can find the claimed Ethereum address.
+Obtain the tip of its identity chain and read the Ethereum address from the identity doc
+`address` field in section `https://radicle.xyz/ethereum/claim/v1`, unless it's expired.
+You need to verify that the given Ethereum address claims back the Link identity root,
+see [Discovery from an Ethereum address](#discovery-from-an-ethereum-address).
```

Comment:
suggestion: Have a high-level "Discovery" section with two sub-sections "From an Ethereum address" and "From a Radicle Link Identity"

Reply:
Fixed